### PR TITLE
Implement base64 encoding and decoding functions

### PR DESCRIPTION
**Pull request by elprans**
This adds the new `std::encoding::base64_encode` and
`std::encoding::base64_decode` functions to encode to and decode from
base64-encoded strings.  There is support for RFC 4648 §4 standard
alphabet as well as RFC 4648 §5 URL- and filename-safe alphabet.
Padding requirement can be controlled by the `padding` named only
argument.

### CODE REVIEW

**Comment by 1st1** [comment]
I like "enc" as the name of the new namespace better. Would work better with existing abbreviations like "cal"

File: tests/test_edgeql_functions.py [hunk 1]
```diff
@@ -7476,3 +7478,77 @@ async def test_edgeql_functions_multirange_unpack_01(self):
                 ],
                 json_only=True,
             )
+
+    async def test_edgeql_functions_encoding_base64_fuzz(self):
```
Comment (1st1) [inline tests/test_edgeql_functions.py:7482]: Please add a test to see how gracefully it fails if you try to decode a non-base64 garbage. I'm curious if stray NULLs can appear or unexpected errors.

Comment (1st1) [reply tests/test_edgeql_functions.py:7482]: Also test passing an empty string.

Comment (elprans) [reply tests/test_edgeql_functions.py:7482]: Empty strings are already tested: `random.randrange(0, 1000)`

Comment (elprans) [reply tests/test_edgeql_functions.py:7482]: Done.

**Comment by elprans** [comment]
> I like "enc" as the name of the new namespace better. Would work better with existing abbreviations like "cal"

Changed.

**Review by 1st1 — APPROVED** [review]
